keyboards/theone/nuphy_krypton75/nuphy_krypton75.c

```diff
@@ -0,0 +1,633 @@
+        set_right_rgb(r_temp, g_temp, b_temp);
+        rgb_matrix_update_pwm_buffers();
+        wait_ms(200);
+
+        set_right_rgb(0, 0, 0);
+        rgb_matrix_update_pwm_buffers();
+        wait_ms(200);
+    }
+}
+
+void device_reset_init(void) {
+    side_mode       = 0;
+    side_light      = 40;
+    side_speed      = 3;
+    side_rgb        = 1;
+    side_colour     = 0;
+    side_play_cnt   = 0;
+    side_play_point = 0;
+    side_play_timer = timer_read32();
+
+    rgb_matrix_enable();
+    rgb_matrix_mode(RGB_MATRIX_CYCLE_LEFT_RIGHT);
+    rgb_matrix_set_speed(255 - RGB_MATRIX_SPD_STEP * 2);
+    rgb_matrix_sethsv(255, 255, RGB_MATRIX_MAXIMUM_BRIGHTNESS - RGB_MATRIX_VAL_STEP * 2);
+
+    default_layer_set(1 << 0);
+    keymap_config.nkro        = 0;
+    user_config.ee_layer_mode = LAYER_MAC_MODE;
+
+    user_config.ee_init_flag   = 0xA5;
+    user_config.ee_side_mode   = side_mode;
+    user_config.ee_side_light  = side_light;
+    user_config.ee_side_speed  = side_speed;
+    user_config.ee_side_rgb    = side_rgb;
+    user_config.ee_side_colour = side_colour;
+    eeconfig_update_user_datablock(&user_config);
+}
+
+void long_press_key(void) {
+    static uint32_t long_press_timer = 0;
+    static uint16_t reset_press_time = 0;
+    static uint16_t layer_sw_press_time;
+
+    if (timer_elapsed32(long_press_timer) < 100) return;
+    long_press_timer = timer_read32();
+
+    if (f_layer_mac_press || f_layer_win_press) {
+        layer_sw_press_time++;
+        if (layer_sw_press_time >= 30) {
+            layer_sw_press_time = 0;
+
+            if (f_layer_mac_press) {
+                default_layer_set(1 << 0);
+                keymap_config.nkro        = 0;
+                user_config.ee_layer_mode = LAYER_MAC_MODE;
+            } else if (f_layer_win_press) {
+                default_layer_set(1 << 2);
+                keymap_config.nkro        = 1;
+                user_config.ee_layer_mode = LAYER_WIN_MODE;
+            }
+            f_layer_win_press = 0;
+            f_layer_mac_press = 0;
+
+            eeconfig_update_user_datablock(&user_config);
+
+            device_sys_show();
+        }
+    } else {
+        layer_sw_press_time = 0;
+    }
+
+    if (f_reset_press) {
+        reset_press_time++;
+        if (reset_press_time >= 30) {
+            reset_press_time = 0;
+
+            f_reset_press = 0;
+            eeconfig_init();
+
+            device_reset_show();
+
+            device_reset_init();
+        }
+    } else {
+        reset_press_time = 0;
+    }
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+    switch (keycode) {
+        case MAC_TASK: {
+            if (record->event.pressed) {
+                host_consumer_send(0x029F);
+            } else {
+                host_consumer_send(0);
+            }
+            return false;
+        }
```

Comment:
This is just `KC_MISSION_CONTROL` but with more steps.

Reply:
Thank you, I have adjusted it